realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1269,6 +1278,45 @@ public static Object getDefaultModule() {
     }
 
     /**
+     * Acquire a reference to the given Realm file.
+     * This function call should be called in a synchronized block on BaseRealm.class
+     *
+     * @return the reference count after acquiring.
+     */
+    private static int acquireRealmFileReference(RealmConfiguration configuration) {
+        String path = configuration.getPath();
+        Integer refCount = typedRealmFileReferenceCounter.get(path);
+        if (refCount == null) {
+            refCount = 0;
+        }
+        refCount += 1;
+        typedRealmFileReferenceCounter.put(path, refCount);
+        return refCount;
+    }
+
+    /**
+     * Releases a reference to the Realm file. If reference count reaches 0 any cached configurations
+     * will be removed.
+     * This function call should be called in a synchronized block on BaseRealm.class
+     *
+     * @return the reference count after releasing.
+     */
+    private static int releaseRealmFileReference(RealmConfiguration configuration) {
+        String path = configuration.getPath();
+        Integer refCount = typedRealmFileReferenceCounter.get(path);
+        if (refCount == null || refCount <= 0) {
```

Comment:
Is this scenario possible? It is called from `lastLocalInstanceClosed()` which has it's own counter check in BaseRealm?